api/replica_test.go

```diff
@@ -0,0 +1,101 @@
+package api
+
+import (
+	// weird imports for testing
+	"net/http/httptest"
+	"sync/atomic"
+	"testing"
+
+	// we need to import context under a different name
+	gocontext "context"
```

Comment:
Why do we need to alias `context`?

Reply:
because it's a type in this package as well. or a global variable maybe. go vomitted errors, i'd have to double check what error it was.

Reply:
Swarm defines its own `context` struct (this was before contexts were a thing in Golang), which might explain the errors.

Reply:
there's a type `context` declared in `primary.go`, toward the top.

Reply:
I realize this is a bit of a pain, but can you just rename the struct in `primary.go` instead? It'll make this code more readable.

Reply:
I'm fine with doing that, but it exists in a lot of places, and might just make this PR harder to review/unnecessarily cluttered. Maybe let's do it as a follow up?

Reply:
All right, let's do it in a follow-up PR.